proposals/p1983.md

```diff
@@ -0,0 +1,183 @@
+
+-   For decimal integers, the digit separators shall occur every three digits
+    starting from the right. For example, `2_147_483_648`.
+-   For hexadecimal integers, the digit separators shall occur every four digits
+    starting from the right. For example, `0x7FFF_FFFF`.
+-   For real number literals, digit separators can appear in the decimal and
+    hexadecimal integer portions (prior to the period and after the optional `e`
+    or mandatory `p`) as described in the previous bullets. For example,
+    `2_147.483648e12_345` or `0x1_00CA.FEF00Dp+24`
+-   For binary literals, digit separators can appear between any two digits. For
+    example, `0b1_000_101_11`.
+
+Some issues with this approach have been raised:
+
+-   For integers, while the original proposal mentioned Indian digit groupings,
+    Chinese, Japanese, and Korean cultures use 4-digit groupings. This oversight
+    is likely due to the description on
+    [wikipedia](https://en.wikipedia.org/wiki/Decimal_separator#Digit_grouping):
+    "but the delimiter commonly separates every three digits".
+-   For hexadecimal in particular, there are microformats where different
+    placement rules may be desirable. For example, MAC addresses are typically
+    in byte pairs (`00-B0-D0-63-C2-26`) and UUIDs have particular groupings
+    (`123e4567-e89b-12d3-a456-426614174000`).
+
+This was asked on
+[Issue #1485: Reconsider digit separators](https://github.com/carbon-language/carbon-lang/issues/1485),
+where a proposal was requested.
+
+## Proposal
+
+Switch to simple rules: a single digit separator can appear between any two
+digits. For example:
+
+-   Decimal literals: `1_2_3`
+-   Hexadecimal literals: `0xA_B_C`
+-   Real literals: `1_2.3_4e5_6`
+-   Binary literals are unaffected.
+
+## Rationale
+
+-   [Community and culture](/docs/project/goals.md#community-and-culture)
+    -   Removing restrictions on decimal literals provides flexibility for
+        developers who don't use 3-digit groupings.
+-   [Code that is easy to read, understand, and write](/docs/project/goals.md#code-that-is-easy-to-read-understand-and-write)
+    -   This allows for mistakes in groupings, which is undesirable. However,
+        it's useful for microformats to be able to provide their own particular
+        groupings.
+-   [Interoperability with and migration from existing C++ code](/docs/project/goals.md#interoperability-with-and-migration-from-existing-c-code)
+    -   Reduces migration hurdles for C++ code by providing a rule that's more
+        consistent.
+
+## Alternatives considered
+
+### 3-digit decimal groupings
+
+For decimal separators, we could enforce 3-digit groupings if digit separators
+are used at all.
+
+Advantages:
+
+-   More enforcement of digit groupings can prevent bugs and misreadings
+    resulting from accidentally incorrect groupings.
+
+Disadvantages:
+
+-   Indian, Chinese, Japanese, and Korean digit grouping conventions don't fit
+    under this rule.
```

Comment:
I think it is useful to draw a direct analogy to the benefits of allowing unicode characters in identifiers so that folks using a language with a non-Latin alphabet can directly use their language in the identifiers.

Reply:
Done